### PR TITLE
[CI] enable dependabot

### PR DESCRIPTION
**Pull request by jabia99**
dependabot is an independent security scanning tool which mostly focuses on evaluating the dependency chain. Having the dependabot.yml file on the main branch would enable the bot to test the dependencies and create a pull request for each one, on a specified schedule.

When we finish dependabot configuration tweaking in osbuild-composer repository, we can merge this PR.

### CODE REVIEW

**Comment by Gundersanne** [comment]
CI got fixed in #245, needs a rebase to unbreak CI

**Comment by ondrejbudai** [comment]
I have a couple of thoughts after I tried to get as many Dependabot PRs as possible merged over the past weekend:

- I think it's a good idea to mass-update dependencies manually before enabling Dependabot. It greatly reduces stress on our resources.
- PRs from Dependabot very often created conflicts in `go.sum`. If this becomes an issue even after the mass-update, we may want to reduce the number of open PRs to 1 because it's basically a game of rebasing otherwise. I would definitely start with 5, watch them closely and react accordingly.

**Comment by jabia99** [comment]
Okay, we can merge this once the manual dependency update PR is merged.

**Comment by ondrejbudai** [comment]
> Okay, we can merge this once the manual dependency update PR is merged.

Are you working on such a PR?

**Comment by jabia99** [comment]
> Are you working on such a PR?

I wouldn't know were to start without enabling dependabot. I've been taking a look at this  [combine-dpendabot-prs](https://www.npmjs.com/package/combine-dependabot-prs). But I don't know if it's worth the effort.

File: .github/dependabot.yml
Comment (Gundersanne) [inline .github/dependabot.yml:19]: I know composer has the same config; but 5 PRs which all change `go.mod` sounds like a recipe for rebase issues. I wonder if it'll be more user-friendly if it has a limit of 1 open PR, but then checks every 4 hours.

Comment (jabia99) [reply .github/dependabot.yml:19]: I don't think changing the schedule is needed. From [this](https://github.com/osbuild/osbuild-composer/pull/1724#issuecomment-913013581) comment from Ondřej, I guess dependabot stores the PR on a queue whenever there are more the the configured limit.

I'll change the open PR limit now.

**Review by Gundersanne — APPROVED** [review]

**Comment by Gundersanne** [comment]
@jabia99 Thank you!

 For the commit message format, while not required, https://www.osbuild.org/guides/developer-guide/workflow.html#commits I really like this format.

No need to force push again, I'll edit the commit message on squash merging. But something for future commits.

**Comment by jabia99** [comment]
Great, thanks for the tip!!

**Pull request merged**
